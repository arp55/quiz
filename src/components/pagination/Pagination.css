.page-dot {
  height: 14px;
  width: 14px;
  border-radius: 10px;
  margin-right: 10px;
  background-color: gray;
  margin-left: 10px;
  cursor: pointer;
}

.icon-page {
  height: 20px;
  width: 20px;
  cursor: pointer;
}

.dot-cont {
  display: flex;
  align-items: center;
}

.focused {
  opacity: 0.5;
}
