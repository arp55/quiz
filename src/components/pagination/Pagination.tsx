import React, { useState } from 'react'
import './Pagination.css'
import { ChevronLeft, ChevronRight } from '@mui/icons-material';
import { quizData } from '../../quizData';

type Props = {
    setChange: Function
}

export default function Pagination({ setChange }: Props) {
    const [page, setpage] = useState(1)

    function handleClick(val: string) {
        if (val === 'left' && page > 1) {
            setpage(page => page - 1)
            setChange(page - 1)
        }
        if (val === 'right' && page < quizData.length) {
            setpage(page => page + 1)
            setChange(page + 1)
        }
    }

    function changePage(index: number) {
        setpage(page => index + 1)
        setChange(index + 1)
    }

    return (
        <div className='dot-cont'>
            <ChevronLeft data-testid='chevron-icon' className="icon-page" onClick={() => handleClick('left')} />
            {new Array(quizData.length).fill(null).map((_, index: number) => {
                let className = 'page-dot'
                if (page === index + 1) {
                    className += ' focused'
                }
                return (
                    <>
                        <div onClick={() => changePage(index)} {...{ className }} />
                    </>
                )
            })}
            <ChevronRight className="icon-page" onClick={() => handleClick('right')} />
        </div>
    )
}
