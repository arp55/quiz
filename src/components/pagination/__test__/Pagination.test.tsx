import React from 'react'
import renderer from 'react-test-renderer'
import Pagination from '../Pagination'
import { render } from '@testing-library/react'


describe('pagination tests', () => {
    test('pagination snapshot', () => {
        const tree = renderer.create(<Pagination setChange={() => { }} />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('left icon present', () => {
        const { getByTestId } = render(<Pagination setChange={() => { }} />)
        expect(getByTestId('chevron-icon')).toBeInTheDocument()
    })
})