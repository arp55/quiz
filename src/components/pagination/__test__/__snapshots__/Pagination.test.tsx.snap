// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination tests pagination snapshot 1`] = `
<div
  className="dot-cont"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon-page css-i4bv87-MuiSvgIcon-root"
    data-testid="chevron-icon"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
    />
  </svg>
  <div
    className="page-dot focused"
    onClick={[Function]}
  />
  <div
    className="page-dot"
    onClick={[Function]}
  />
  <div
    className="page-dot"
    onClick={[Function]}
  />
  <div
    className="page-dot"
    onClick={[Function]}
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon-page css-i4bv87-MuiSvgIcon-root"
    data-testid="ChevronRightIcon"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
    />
  </svg>
</div>
`;
